src/mokernels/moinput.jl

```diff
@@ -8,23 +8,12 @@ struct MOInput{T<:AbstractVector} <: AbstractVector{Tuple{Any,Int}}
     out_dim::Integer
 end
 
-Base.length(inp::MOInput) = inp.out_dim * length(inp.x)
-
-Base.size(inp::MOInput, d) = d::Integer == 1 ? inp.out_dim * size(inp.x, 1) : 1 
 Base.size(inp::MOInput) = (inp.out_dim * size(inp.x, 1),)
```

Comment:
To be consistent with the definition of the indexing operation, this should probably be
```suggestion
Base.size(inp::MOInput) = (inp.out_dim * length(inp.x),)
```